src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,36 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate fromDate = LocalDate.parse(dateFrom, formatter);
+        LocalDate toDate = LocalDate.parse(dateTo, formatter);
+        StringBuilder stringBuilder = new StringBuilder("Report for period ");
+        stringBuilder.append(dateFrom)
+                .append(" - ")
+                .append(dateTo);
+        for (String name : names) {
+            int salary = 0;
+            for (int i = 0; i < data.length; i++) {
+                String[] slrInfo = data[i].split(" ");
+                LocalDate slrDate = LocalDate.parse(slrInfo[0], formatter);
+                String employerName = slrInfo[1];
+                int hoursOfWork = Integer.parseInt(slrInfo[2]);
+                int salaryPerHour = Integer.parseInt(slrInfo[3]);
```

Comment:
These indexes better make constants

Reply:
> These indexes better make constants
But how they can be constants?  If i need to change them dynamic